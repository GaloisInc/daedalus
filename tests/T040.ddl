def Main =
  { a = ^ true;
    b = ^ false;
    c = ^ nothing : maybe bool;
    d = ^ just true;
    e = d is nothing <| ^ {};
    f = d is just <| ^ true;
  }

