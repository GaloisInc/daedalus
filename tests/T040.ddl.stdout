module T040
 
--- Imports:
 
--- Type defs:
type T040.Main = { a: bool
                 ; b: bool
                 ; c: Maybe bool
                 ; d: Maybe bool
                 ; e: {}
                 ; f: bool
                 }
 
--- Rules:
 
T040.Main : Grammar T040.Main =
  do (a : bool) <- pure true
     (b : bool) <- pure false
     (c : Maybe bool) <- pure nothing
     (d : Maybe bool) <- pure (just true)
     (e : {}) <- Choose biased
                   { Guard (d == nothing)
                   | pure {}
                   }
     (f : bool) <- Choose biased
                     { d is just
                     | pure true
                     }
     pure {a = a,
           b = b,
           c = c,
           d = d,
           e = e,
           f = f}
 
T040._Main : Grammar {} =
  do (d : Maybe bool) <- pure (just true)
     Choose biased
       { Guard (d == nothing)
       | pure {}
       }
     Choose biased
       { @(d is just)
       | pure {}
       }
